src/mlpack/core/data/types_impl.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file core/data/types.hpp
+ * @author Gopi M. Tatiraju
+ * 
+ * This file contains utilitiy fucntions related to types of data.
+ * We have adapted all the standard types which are available in armadillo.
+ * 
+ * This file also contains functions to convery external file types to mlpack
+ * file types. In future if we the user of mlpack needs support of an external
+ * linear algebra library like armadillo, all fucntions related to handling the
+ * types goes here.
+ *
+ * https://gitlab.com/conradsnicta/armadillo-code/-/blob/10.5.x/include/armadillo_bits/diskio_meat.hpp
+ * Copyright 2008-2016 Conrad Sanderson (http://conradsanderson.id.au)
+ * Copyright 2008-2016 National ICT Australia (NICTA)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------
```

Comment:
I am not sure that we have to put the Apache license here

Reply:
Oh sorry, yea we don't need to. Will remove it.